samples/sample_common/src/sysmem_allocator.cpp

```diff
@@ -392,7 +392,11 @@ mfxStatus SysMemBufferAllocator::AllocBuffer(mfxU32 nbytes, mfxU16 type, mfxMemI
         return MFX_ERR_UNSUPPORTED;
 
     mfxU32 header_size = MSDK_ALIGN32(sizeof(sBuffer));
+#if defined (LINUX32) || defined (LINUX64)
+    mfxU8 *buffer_ptr = (mfxU8 *)aligned_alloc(sysconf(_SC_PAGESIZE), header_size + nbytes + 32);
```

Comment:
Need to zero memory to align with Windows

Reply:
Changed calloc to malloc and set zero memory below (like in libmfx_allocator).

Reply:
I thought Windows gas _aligned_alloc and _aligned_free pair. Can we use them and introduce some high level macro  covering both windows and linux? Like:
```
#ifdef _WIN32
#define ALIGNED_ALLOC(X, S) _aligned_malloc(X, S)
#define ALIGNED_ALLOC(P) _aligned_free(P)
#else 
#define ALIGNED_ALLOC(X, S) aligned_malloc(X, S)
#define ALIGNED_ALLOC(P) free(X, S)
#endif 

```
Or we can try follow @mgonchar proposal to use std.

Reply:
Also, let's follow our [coding guide](https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/Coding_guidelines.md#memory-management) and switch to c++ cast at least in new code :)